src/Utils.jl

```diff
@@ -1,6 +1,20 @@
 # Utility functions, which can be used by different top level scripts
 
 using CodecZlib
+using CodecZstd
+using Match
+
+"""
+    open_with_stream(fname::AbstractString)
+
+Open a file with a stream decompressor if it is compressed with gzip or zstd,
+otherwise as a normal file.
+"""
+open_with_stream(fname::AbstractString) = @match fname begin
+    r"\.gz$" => GzipDecompressorStream(open(fname))
```

Comment:
do we need `@match`, or do we just want to check
```julia
endswith(fname, ".gz")
```

Reply:
That is the way that the original code was, but I have an aversion to nested `if` `elseif` blocks when `@match` is more compact (at heart, I am hankering for `switch:`). This should be a very optimised regex when compiled, so I'm not worried about performance.

Reply:
One may even consider adding more stream readers, thought it's probably fine with just these two

Reply:
`Match` is a bit OTT for this, so I reimplemented with a few nested `if`s